libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Conway/LedgerTypes/Specs.hs

```diff
@@ -213,17 +214,29 @@ aggregateDRep m = Map.foldlWithKey accum Map.empty m
 dstateSpec ::
   forall era fn.
   EraSpecLedger era fn =>
+  Term fn (Set (Credential 'DRepRole (EraCrypto era))) ->
   Term fn AccountState ->
   Term fn (Map (KeyHash 'StakePool (EraCrypto era)) (PoolParams (EraCrypto era))) ->
   Specification fn (DState era)
-dstateSpec acct poolreg = constrained $ \ [var| ds |] ->
+dstateSpec drepRoleCredSet acct poolreg = constrained $ \ [var| ds |] ->
   match ds $ \ [var|umap|] [var|futureGenDelegs|] [var|genDelegs|] [var|irewards|] ->
     match umap $ \ [var|rdMap|] [var|ptrmap|] [var|sPoolMap|] [var|dRepMap|] ->
-      [ satisfies dRepMap TrueSpec
-      , -- This is passed to vstateSpec to enforce that the random set of DReps
-        -- delegated to actually exist and are registered, and that credentials appear as delegatees
-        -- TODO, see that more than one credential delegates to the same DRep
-        genHint 5 sPoolMap
+      [ -- This field, dRepMap, is passed to vstateSpec to enforce that the set of DReps
+        -- delegated to actually are registered and appear in the DState, and that every credential
+        -- delegated to a DRep, appears in the set of credentials for that DRep.
+        -- The dRepMap depends on the rdMap, so it is computed afterwards, forced by the reify
+        reify rdMap id $ \ [var|rdm|] ->
+          [ assert $ subset_ (dom_ dRepMap) (dom_ rdm)
+          , -- , assert $ sizeOf_ (dom_ dRepMap) >=. lit 1
+            forAll dRepMap $ \ [var|pair|] ->
```

Comment:
This is where things were over constrained. There are rare but valid cases where the Drepmap can be empty.